funcx_endpoint/funcx_endpoint/endpoint/endpoint_manager.py

```diff
@@ -0,0 +1,508 @@
+        log.info("Shutdown complete.\n---------- Endpoint Manager ends ----------\n\n")
+
+    def _event_loop(self):
+        self._command.start()
+
+        local_user_lookup = {}
+        try:
+            with open("local_user_lookup.json") as f:
+                local_user_lookup = json.load(f)
+        except Exception as e:
+            msg = (
+                f"Unable to load local users ({e.__class__.__name__}) {e}\n"
+                "  Will be unable to respond to any commands; update the lookup file"
+                f" and either restart (stop, start) or SIGHUP ({os.getpid()}) this"
+                " endpoint."
+            )
+            log.error(msg)
+
+        valid_method_name_re = re.compile(r"^[A-Za-z][0-9A-Za-z_]{0,99}$")
+        max_skew_s = 180  # 3 minutes; ignore commands with out-of-date timestamp
+        while not self._time_to_stop:
+            if self._wait_for_child:
+                self.wait_for_children()
+
+            try:
+                _command = self._command_queue.get(timeout=1.0)
+                d_tag, props, body = _command
+                if props.headers and props.headers.get("debug", False):
+                    body_log_b = re.sub(rb"://([^:]+):[^@]+@", rb"://\1:***@", body)
+                    log.warning(
+                        "Command debug requested:"
+                        f"\n  Delivery Tag: {d_tag}"
+                        f"\n  Properties: {props}"
+                        f"\n  Body bytes: {body_log_b}"
+                    )
+            except queue.Empty:
+                if self._command_stop_event.is_set():
+                    self._time_to_stop = True
+                if sys.stderr.isatty():
+                    print(
+                        f"\r{time.strftime('%c')}", end="", flush=True, file=sys.stderr
+                    )
+                continue
+
+            try:
+                server_cmd_ts = props.timestamp
+                if props.content_type != "application/json":
+                    raise ValueError("Invalid message type; expecting JSON")
+
+                msg = json.loads(body)
+                command = msg.get("command")
+                command_args = msg.get("args", [])
+                command_kwargs = msg.get("kwargs", {})
+            except Exception as e:
+                log.error(
+                    f"Unable to deserialize Globus Compute services command."
+                    f"  ({e.__class__.__name__}) {e}"
+                )
+                self._command.ack(d_tag)
+                continue
+
+            now = round(time.time())
+            if abs(now - server_cmd_ts) > max_skew_s:
+                server_pp_ts = datetime.fromtimestamp(server_cmd_ts).strftime("%c")
+                endp_pp_ts = datetime.fromtimestamp(now).strftime("%c")
+                log.warning(
+                    "Ignoring command from server"
+                    "\nCommand too old or skew between system clocks is too large."
```

Comment:
Another log_extra suggestion (more overall formatting).  Unless you intentionally use the line breaks to make it easier to read in logs, which is a value add.  (probably worth slack discussion)